src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/network/v4_9/CloudStackNetworkPluginTest.java

```diff
@@ -2,299 +2,421 @@
+        CloudStackUser cloudStackUser = CloudstackTestUtils.CLOUD_STACK_USER;
 
-        PowerMockito.mockStatic(CloudStackUrlUtil.class);
-        PowerMockito.when(CloudStackUrlUtil.createURIBuilder(Mockito.anyString(), Mockito.anyString())).thenCallRealMethod();
+        NetworkInstance networkInstanceExpected = Mockito.mock(NetworkInstance.class);
+        Mockito.doReturn(networkInstanceExpected).when(this.plugin).doGetInstance(
+                Mockito.any(GetNetworkRequest.class), Mockito.eq(cloudStackUser));
 
-        NetworkOrder networkOrder = new NetworkOrder();
-        networkOrder.setInstanceId(FAKE_ID);
+        GetNetworkRequest request = new GetNetworkRequest.Builder()
+                .id(instanceIdExpected)
+                .build(this.cloudStackUrl);
 
         // exercise
-        this.plugin.deleteInstance(networkOrder, FAKE_CLOUD_USER);
+        NetworkInstance networkInstance = this.plugin.getInstance(networkOrder, cloudStackUser);
 
         // verify
-        PowerMockito.verifyStatic(CloudStackUrlUtil.class, VerificationModeFactory.times(1));
-        CloudStackUrlUtil.sign(Mockito.any(URIBuilder.class), Mockito.anyString());
+        Assert.assertEquals(networkInstanceExpected, networkInstance);
+        RequestMatcher<GetNetworkRequest> matcher = new RequestMatcher.GetNetwork(request);
+        Mockito.verify(this.plugin, Mockito.times(TestUtils.RUN_ONCE)).doGetInstance(
+                Mockito.argThat(matcher), Mockito.eq(cloudStackUser));
+    }
+
+    // test case: When calling the doGetInstance method with secondary methods mocked,
+    // it must verify if It returns the right NetworkInstance.
+    @Test
+    public void testDoGetInstanceSuccessfully() throws FogbowException, HttpResponseException {
+        // set up
+        CloudStackUser cloudStackUser = CloudstackTestUtils.CLOUD_STACK_USER;
+        GetNetworkRequest request = new GetNetworkRequest.Builder().build("");
+        String uriRequestExpected = request.getUriBuilder().toString();
+
+        String responseStr = "anyString";
+        Mockito.when(this.client.doGetRequest(Mockito.eq(uriRequestExpected), Mockito.eq(cloudStackUser)))
+                .thenReturn(responseStr);
 
-        Mockito.verify(this.client, Mockito.times(1)).doGetRequest(expectedRequestUrl, FAKE_CLOUD_USER);
+        GetNetworkResponse response = Mockito.mock(GetNetworkResponse.class);
+        PowerMockito.mockStatic(GetNetworkResponse.class);
+        PowerMockito.when(GetNetworkResponse.fromJson(Mockito.eq(responseStr))).thenReturn(response);
+
+        NetworkInstance networkInstaceExpexted = Mockito.mock(NetworkInstance.class);
+        Mockito.doReturn(networkInstaceExpexted).when(this.plugin).getNetworkInstance(Mockito.eq(response));
+
+        // exercise
+        NetworkInstance networkInstance = this.plugin.doGetInstance(request, cloudStackUser);
+
+        // verify
+        Assert.assertEquals(networkInstaceExpexted, networkInstance);
```

Comment:
check if CloudStackCloudUtils.doGet() has been called too. Please, take a look in the other tests and verify if you're not missing some calls' verification.

Reply:
Backing to the commentary, the test already makes the verify. Look at this [line](https://github.com/fogbow/resource-allocation-service/blob/cloudstack-network-test-refator/src/test/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/network/v4_9/CloudStackNetworkPluginTest.java#L255). 
Is it necessary to put the verify in the end? 
This will double the checking but will make explicit. 
We could debate this.

Reply:
Why this line guarantees it? Isnt the line just mocking the method in case of it get called?

Reply:
once it is fixed or you confirm that the line also guarantees the call it LGTM